tests/pipeline/test_pipeline.py

```diff
@@ -5,23 +5,33 @@
 from snapshottest.module import SnapshotTest
 from typer.testing import CliRunner
 
+import kpops
 from kpops.cli.main import app
 from kpops.pipeline_generator.pipeline import ParsingException
 
 runner = CliRunner()
 
 RESOURCE_PATH = Path(__file__).parent / "resources"
-PIPELINE_BASE_DIR = str(RESOURCE_PATH.parent)
+PIPELINE_BASE_DIR_PATH = RESOURCE_PATH.parent
 
 
 class TestPipeline:
+    def test_python_api(self):
+        pipeline = kpops.generate(
+            RESOURCE_PATH / "first-pipeline" / "pipeline.yaml",
+            "tests.pipeline.test_components",
+            pipeline_base_dir=PIPELINE_BASE_DIR_PATH,
+            defaults=RESOURCE_PATH,
+        )
+        assert len(pipeline) == 3
+
     def test_load_pipeline(self, snapshot: SnapshotTest):
         result = runner.invoke(
             app,
             [
                 "generate",
                 "--pipeline-base-dir",
-                PIPELINE_BASE_DIR,
+                str(PIPELINE_BASE_DIR_PATH),
```

Comment:
Any reason to move the str to the reference and remove it from the definition ?

Reply:
for the new test which needs a `Path` object. besides, it looks a little more uniform with the rest of the arguments